third_party/terraform/resources/resource_container_cluster.go.erb

```diff
@@ -814,6 +814,11 @@ func resourceContainerClusterCreate(d *schema.ResourceData, meta interface{}) er
 	timeoutInMinutes := int(d.Timeout(schema.TimeoutCreate).Minutes())
 	waitErr := containerOperationWait(config, op, project, location, "creating GKE cluster", timeoutInMinutes)
 	if waitErr != nil {
+		if deleteErr := cleanFailedContainerCluster(d, meta); deleteErr != nil {
```

Comment:
Is there a chance a cluster could be created, return an error but still be in a usable state? As in would we want to just preserve the id in state and let the user manage deletion/recreation on their own?

Reply:
IDK about useable state (delete-recreate isn't really what I'd call useable, and I don't think you can just update the cluster to be correct) but I'm not sure about preserving ID in state on failure.